civihr_employee_portal/views/views_export/views_mydetails_myaddress.inc

```diff
@@ -0,0 +1,228 @@
+<?php
+
+$view = new view();
+$view->name = 'mydetails_myaddress';
+$view->description = '';
+$view->tag = 'default';
+$view->base_table = 'civicrm_contact';
+$view->human_name = 'MyDetails_MyAddress';
+$view->core = 7;
+$view->api_version = '3.0';
+$view->disabled = FALSE; /* Edit this to true to make a default view disabled initially */
+
+/* Display: Master */
+$handler = $view->new_display('default', 'Master', 'default');
+$handler->display->display_options['title'] = 'Home Address';
+$handler->display->display_options['css_class'] = 'chr_panel--my-details__view__panel';
+$handler->display->display_options['use_more_always'] = FALSE;
+$handler->display->display_options['access']['type'] = 'perm';
+$handler->display->display_options['access']['perm'] = 'view my details';
+$handler->display->display_options['cache']['type'] = 'none';
+$handler->display->display_options['query']['type'] = 'views_query';
+$handler->display->display_options['exposed_form']['type'] = 'basic';
+$handler->display->display_options['pager']['type'] = 'some';
+$handler->display->display_options['pager']['options']['items_per_page'] = '1';
+$handler->display->display_options['pager']['options']['offset'] = '0';
+$handler->display->display_options['style_plugin'] = 'default';
+$handler->display->display_options['row_plugin'] = 'fields';
+$handler->display->display_options['row_options']['hide_empty'] = TRUE;
+/* Footer: Global: Text area */
+$handler->display->display_options['footer']['area']['id'] = 'area';
+$handler->display->display_options['footer']['area']['table'] = 'views';
+$handler->display->display_options['footer']['area']['field'] = 'area';
+$handler->display->display_options['footer']['area']['content'] = '<a href="/edit-my-home-address/js/view" class="ctools-use-modal ctools-modal-civihr-custom-style  btn btn-primary">
+  <i class="fa fa-pencil" aria-hidden="true"></i>
+    Edit my address
+</a>
+';
+$handler->display->display_options['footer']['area']['format'] = 'html_without_filters';
+/* No results behavior: Global: Text area */
+$handler->display->display_options['empty']['area']['id'] = 'area';
+$handler->display->display_options['empty']['area']['table'] = 'views';
+$handler->display->display_options['empty']['area']['field'] = 'area';
+$handler->display->display_options['empty']['area']['empty'] = TRUE;
+$handler->display->display_options['empty']['area']['content'] = '<p class="view-empty-message">No home address listed!</p>
```

Comment:
We have the same structure for Personal Details also. Can we move this to a tpl file and use variables to change `href` and `label`?

Reply:
@deb1990 this is exported code from a view, so while it is doable what you are asking because it is code it must be handled as auto-generated code (because it is). Also, footer and empty sections on a view are editable, and Drupal expects and will output only HTML in this section, making useless any PHP.

Reply:
Ok lets keep as it is right now.